core/providers/ruby/ruby.go

```diff
@@ -0,0 +1,392 @@
+}
+
+func (p *RubyProvider) Build(ctx *generate.GenerateContext, build *generate.CommandStepBuilder) []string {
+	build.Secrets = []string{}
+	build.UseSecretsWithPrefixes([]string{"RAILS", "BUNDLE", "BOOTSNAP", "SPROCKETS", "WEBPACKER", "ASSET", "DISABLE_SPRING"})
+	build.AddEnvVars(p.GetRubyEnvVars(ctx))
+	build.AddCommand(plan.NewCopyCommand("."))
+	outputs := []string{"/app"}
+	// Only compile assets if a Rails app have an asset pipeline gem
+	// installed (e.g. sprockets, propshaft). Rails API-only apps [0]
+	// do not come with the asset pipelines because they have no assets.
+	// [0] https://guides.rubyonrails.org/api_app.html
+	if p.usesRails(ctx) && p.usesAssetPipeline(ctx) {
+		build.AddCommand(plan.NewExecCommand("bundle exec rake assets:precompile"))
+	}
+
+	if p.usesRails(ctx) && p.usesDep(ctx, "bootsnap") {
+		build.AddCommand(plan.NewExecCommand("bundle exec bootsnap precompile app/ lib/"))
+		outputs = append(outputs, "lib/")
+	}
+
+	return outputs
+}
+
+func (p *RubyProvider) GetImageWithRuntimeDeps(ctx *generate.GenerateContext) plan.Input {
+	packages := []string{"libyaml-dev"}
+
+	if p.usesPostgres(ctx) {
+		packages = append(packages, "libpq-dev")
+	}
+
+	if p.usesMysql(ctx) {
+		packages = append(packages, "default-libmysqlclient-dev")
+	}
+
+	if p.usesDep(ctx, "magick") {
+		packages = append(packages, "libmagickwand-dev")
+	}
+
+	if p.usesDep(ctx, "vips") {
+		packages = append(packages, "libvips-dev")
+	}
+
+	if p.usesDep(ctx, "charlock_holmes") {
+		packages = append(packages, "libicu-dev", "libxml2-dev", "libxslt-dev")
+	}
+
+	return ctx.DefaultRuntimeInputWithPackages(packages)
+}
+
+func (p *RubyProvider) GetBuilderDeps(ctx *generate.GenerateContext) *generate.MiseStepBuilder {
+	miseStep := ctx.GetMiseStepBuilder()
+	miseStep.SupportingAptPackages = append(miseStep.SupportingAptPackages, "procps")
+
+	if p.usesPostgres(ctx) {
+		miseStep.SupportingAptPackages = append(miseStep.SupportingAptPackages, "libpq-dev")
+	}
+
+	if p.usesMysql(ctx) {
+		miseStep.SupportingAptPackages = append(miseStep.SupportingAptPackages, "default-libmysqlclient-dev")
+	}
+
+	return miseStep
+}
+
+func (p *RubyProvider) InstallMisePackages(ctx *generate.GenerateContext, miseStep *generate.MiseStepBuilder) {
+	ruby := miseStep.Default("ruby", DEFAULT_RUBY_VERSION)
+
+	if envVersion, varName := ctx.Env.GetConfigVariable("RUBY_VERSION"); envVersion != "" {
+		miseStep.Version(ruby, envVersion, varName)
+	}
+
+	if versionFile, err := ctx.App.ReadFile(".ruby-version"); err == nil {
+		miseStep.Version(ruby, utils.ExtractSemverVersion(string(versionFile)), ".ruby-version")
+	}
+
+	if gemfileVersion := parseVersionFromGemfile(ctx); gemfileVersion != "" {
+		miseStep.Version(ruby, gemfileVersion, "Gemfile")
+	}
+
+	miseStep.AddSupportingAptPackage("libyaml-dev")
+	version := p.getRubyVersion(ctx)
+	version = utils.ExtractSemverVersion(version)
+	semver, err := utils.ParseSemver(version)
+	// YJIT in Ruby 3.1+ requires rustc to install
```

Comment:
wat

Reply:
🙃 YJIT is a rust based JIT that Shopify has been driving. It replaces the internal one written in C. It's good that we support it because it's becoming more adopted.